pkgs/applications/graphics/inkscape/extensions/silhouette/default.nix

```diff
@@ -0,0 +1,78 @@
+{ fetchFromGitHub
+, lib
+, pkgs
+, python3
+, umockdev
+, writeScript
+}:
+
+let
+  launch-sendto_silhouette = writeScript "sendto_silhouette.sh" ''
+    cd $(dirname $0)
+    ./sendto_silhouette.py "$@"
+  '';
+  launch-silhouette_multi = writeScript "silhouette_multi.sh" ''
+    cd $(dirname $0)
+    ./silhouette_multi.py "$@"
+  '';
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "inkscape-silhouette";
+  version = "1.28.pre+unstable=2023-06-03"; # Note: there has been lot of as-yet-not-released development since 1.27 was released: https://github.com/fablabnbg/inkscape-silhouette/releases/tag/v1.27
+
+  src = fetchFromGitHub {
+    owner = "fablabnbg";
+    repo = pname;
+    rev = "a1941b6f740466a20dfa116415dd599b1e496302";
+    sha256 = "sha256-CLmSHIhaqtSzPXB1HgHPG+fRW7ktcN3Y3IEWuCVLarw=";
+  };
+
+  patches = [
+    ./interpreter.patch
+  ];
+
+  propagatedBuildInputs = [
+    python3.pkgs.pyusb
+    python3.pkgs.lxml
+    python3.pkgs.inkex
+    python3.pkgs.matplotlib
+  ];
+
+  nativeCheckInputs = [
+    python3.pkgs.pytestCheckHook
+    umockdev
+  ];
+
+  pytestFlagsArray = [
+    "test"
+  ];
+
+  format = "setuptools";
+
+  doCheck = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/inkscape/extensions
+    make install DESTDIR=$out PREFIX=
```

Comment:
It would probably be saner to just use `PREFIX` and patch udevdir to use PREFIX (ideally upstream). See https://github.com/NixOS/nixpkgs/issues/65718. Then we could use the default `installPhase` and pass the path with:

```nix
makeFlags = [
  "PREFIX=${placeholder "out"}"
];
```

Reply:
Neat idea! A couple of thoughts:

1. I've added a patch to get their `Makefile` to use `PREFIX` and changed things to just use the default `installPhase`. However, I'm not optimistic about upstreaming it (because it would change the default value for `UDEV` from `/lib/udev` to `/usr/lib/udev`. Did you have something else in mind?
2. In order to use the default `installPhase` with mk-python-derivation.nix, it looks like I have to either set `format = "other";`, or specify `dontUsePipInstall = true;`. I opted for the `dontUsePipInstall` approach.

Reply:
@jtojnar I'm going to mark this discussion resolved, but feel free to reopen it if you're not happy with the decisions I made!

Reply:
For the record, using `dontUsePipInstall` stopped working due to https://github.com/nixos/nixpkgs/commit/6c85fff302615c62bf4f632bca661bc48298b0a3, which introduces a new pypaInstallHook without a corresponding dont* option.

It still feels weird to me to change the format to "other" when I know this is a setuptools project, so I've opted to override the buildPhase to call `make` again.

Happy to tackle this some other way if you prefer.